### DISCUSSION TITLE
Am I fundamentally misunderstanding the identifier Tauri configuration?

### DISCUSSION THREAD

**Discussion by Almost-Senseless-Coder** [tauri-apps/tauri#9670] [Q&A]
As per [this discussion](https://github.com/tauri-apps/tauri/discussions/5976), Tauri follows the reverse domain notation for app identifiers, as per the `identifier` setting in `tauri.conf.json`.

So an app named `jack-point` by `almost-senseless.tech` would lead to an identifier like so: `tech.almost-senseless.jack-point`.

That used to work splendidly. Since (at least) beta 14 of the `tauri-cli`, though, That leads to this error message when running `cargo tauri android init`:

```
thread 'main' panicked at C:\Users\timjb\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tauri-cli-2.0.0-beta.15\src\mobile\mod.rs:322:6:
called `Result::unwrap()` on an `Err` value: DomainInvalid { domain: "almost-senseless", cause: NotAsciiAlphanumeric { bad_chars: ['-'] } }
```

Setting `identifier` to `tech.almost-senseless` works, but when we open the generated project in Android Studio, its package name is `tech.jack_point`. Renaming the identifier to `tech.almostsenseless.jackpoint` also works and results in the Android namespace `tech.almostsenseless.jack_point`. What I want to see is `tech.almost_senseless.jack_point` for the Android package names, and `tech.almost-senseless.jack-point` otherwise.

Is this a bug or am i fundamentally misunderstanding the `identifier` setting?

**Comment by FabianLars** [comment]
You're not fundamentally misunderstanding it. As far as i know the issue here is just that the team has not decided how to handle platform differences yet. Namingly the main issue is that android only supports `_` but not `-` while ios supports `-` but not `_`.

Anyway, i don't think the people that have to see this will see this here so please instead open a normal issue about this and clearly communicate the solution you want so that the issue doesn't get closed as a os difference problem.

**Reply by loikki** [reply]
The issue was created here: #9707
